tests/sentry/api/endpoints/test_project_keys.py

```diff
@@ -20,6 +24,48 @@ def test_simple(self):
         assert len(response.data) == 1
         assert response.data[0]["public"] == key.public_key
 
+    def test_use_case(self):
+        project = self.create_project()
+        key1 = ProjectKey.objects.get_or_create(project=project)[0]
+        ProjectKey.objects.get_or_create(use_case=UseCase.PROFILING.value, project=project)
+        self.login_as(user=self.user)
+        url = reverse(
+            "sentry-api-0-project-keys",
+            kwargs={"organization_slug": project.organization.slug, "project_slug": project.slug},
+        )
+        response = self.client.get(url)
+        assert response.status_code == 200
+        assert len(response.data) == 1
+        response_data = response.data[0]
+        assert "use_case" not in response_data
+        assert response_data["public"] == key1.public_key
```

Comment:
The linked design doc states that internal DSNs should not show up in public APIs. Should this be a 404 as the user doesn't have an active superuser/staff session.

Reply:
I wanted to test that regular users can see user keys but not internal keys in the list endpoint. I added a comment and better variable names to clarify.